vdyp-core/src/main/java/ca/bc/gov/nrs/vdyp/common/ValueOrMarker.java

```diff
@@ -0,0 +1,123 @@
+package ca.bc.gov.nrs.vdyp.common;
+
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Function;
+
+/**
+ * Type union of a value and an out of band marker.
+ *
+ * @author Kevin Smith, Vivid Solutions
+ *
+ * @param <Value>
+ * @param <Marker>
+ */
+public class ValueOrMarker<Value, Marker> {
+	private final boolean isMarker;
+	private final Object obj;
+
+	private ValueOrMarker(Object obj, boolean isMarker) {
+		Objects.requireNonNull(obj, "The Value or Marker of a ValueOrMarker must not be null");
+
+		this.isMarker = isMarker;
+		this.obj = obj;
+	}
+
+	/**
+	 * Is there a value
+	 *
+	 * @return
+	 */
+	public boolean isMarker() {
+		return isMarker;
+	}
+
+	/**
+	 * Is there a marker
+	 *
+	 * @return
+	 */
+	public boolean isValue() {
+		return !isMarker;
+	}
+
+	/**
+	 * Get the value if present
+	 *
+	 * @return
+	 */
+	@SuppressWarnings("unchecked")
+	public Optional<Value> getValue() {
```

Comment:
Shouldn't getValue throw when the MarkerOrValue is not a value, since Optional.empty() is a possible value()?

Reply:
No, that's why it returns Optional.